Source/NSFontCollection.m

```diff
@@ -127,14 +129,15 @@ + (void) _loadAvailableFontCollections
 /*
  * Writing and Removing Files
  */
-- (BOOL) _writeToFile: (NSString *)path
+- (BOOL) _writeToFile
 {
   NSFileManager *fm = [NSFileManager defaultManager];
   NSString      *tmpPath;
   BOOL          isDir;
   BOOL          success;
   BOOL          path_is_standard = YES;
-
+  NSString     *path = nil;
```

Comment:
Now path will always be nil further down and you may remove most of the non standard logic there.